web/lib/HabitatEmojiPicker.js

```diff
@@ -0,0 +1,1407 @@
+              <span><emoji-purple-square></emoji-purple-square></span>
+              <span><emoji-brown-square></emoji-brown-square></span>
+              <span><emoji-black-large-square></emoji-black-large-square></span>
+              <span><emoji-white-large-square></emoji-white-large-square></span>
+              <span><emoji-black-medium-square></emoji-black-medium-square></span>
+              <span><emoji-white-medium-square></emoji-white-medium-square></span>
+              <span><emoji-black-medium-small-square></emoji-black-medium-small-square></span>
+              <span><emoji-white-medium-small-square></emoji-white-medium-small-square></span>
+              <span><emoji-black-small-square></emoji-black-small-square></span>
+              <span><emoji-white-small-square></emoji-white-small-square></span>
+              <span><emoji-large-orange-diamond></emoji-large-orange-diamond></span>
+              <span><emoji-large-blue-diamond></emoji-large-blue-diamond></span>
+              <span><emoji-small-orange-diamond></emoji-small-orange-diamond></span>
+              <span><emoji-small-blue-diamond></emoji-small-blue-diamond></span>
+              <span><emoji-red-triangle-pointed-up></emoji-red-triangle-pointed-up></span>
+              <span><emoji-red-triangle-pointed-down></emoji-red-triangle-pointed-down></span>
+              <span><emoji-diamond-with-a-dot></emoji-diamond-with-a-dot></span>
+              <span><emoji-radio-button></emoji-radio-button></span>
+              <span><emoji-white-square-button></emoji-white-square-button></span>
+              <span><emoji-black-square-button></emoji-black-square-button></span>
+              <span><emoji-chequered-flag></emoji-chequered-flag></span>
+              <span><emoji-triangular-flag></emoji-triangular-flag></span>
+              <span><emoji-crossed-flags></emoji-crossed-flags></span>
+              <span><emoji-black-flag></emoji-black-flag></span>
+              <span><emoji-white-flag></emoji-white-flag></span>
+      </div>   
+    </div>
+  </div>
+</div>
+
+
+<div id="picker-button" class="flex">
+    <span>
+        <div style='position:absolute;font-size:10px;transform:translate(-50%, -50%);left:50%;bottom:50%;white-space:nowrap;'>Emoji &#x25bc;</div>
+    </span>
+</div>
+
+
+`;
+
+//consts
+
+class HabitatEmojiPicker extends HTMLElement {
+
+   constructor() {
+    super();
+
+    this.attachShadow({ mode: 'open' });
+    this.shadowRoot.append(TEMPLATE.content.cloneNode(true));
+
+    this.pickerBtn = this.shadowRoot.querySelector('#picker-button');
+    this.popover = this.shadowRoot.querySelector('#emoji-popover');
+    this.selector = this.shadowRoot.querySelector('#emoji-select');
+
+    this.pickerBtn.addEventListener('click', (evt) => {
+        evt.stopPropagation();
+        this.popover.classList.toggle('active');
+    });
+
+    for(const elem of this.selector.querySelectorAll('*')) {
```

Comment:
Instead of looping on all the emoji elements we can add one event listener on the parent element and use event delegation. We'll be able to get the option selected by the user by looking at the event target. This should reduce the number of event listeners to just 1 for this component.